node/utils/parseFacetsFromSegment.ts

```diff
@@ -0,0 +1,21 @@
+const parseFacetsFromSegment = (facetsString: string): SelectedFacet[] => {
+  if (!facetsString) {
+    return []
+  }
+
+  // "Remove ";" if it is the last character
+  const cleanFacetsString =
+    facetsString[facetsString.length - 1] === ';'
+      ? facetsString.substring(0, facetsString.length - 1)
+      : facetsString
+
+  return cleanFacetsString.split(';').map(facet => {
```

Comment:
just a suggestion, but if the last character is `;`, you could just filter the empty string after the split :)

```suggestion
  return cleanFacetsString.split(';').filter(facet => facet).map(facet => {
```